File: app/src/main/java/com/example/pertemuan8/ui/view/viewmodel/RencanaStudyViewModel.java
package com.example.pertemuan8.ui.view.viewmodel;

import static kotlinx.coroutines.flow.StateFlowKt.MutableStateFlow;

public class RencanaStudyViewModel : ViewModel() {
    private val_krsState = MutableStateFlow(RencanaStudi())
    val krsState: StateFlow<RencanaStudi> = _krsState.asStateFlow()

    fun setMataKuliah(mkPilihan: String) {
        _krsState.update { stateMK -> stateMK.copy(namaMk = mkPilihan) }
    }
    fun setKelas(kelasPilihan: String) {
        _krsState.update { stateKelas -> stateKelas.copy(kelas = kelasPilihan) }
    }
    fun saveDataKRS(ls: MutableList<String>){
        _krsState.update {
            status -> status.copy(
                    namaMk = ls[0],
                    kelas = ls[1]
            )
        }
    }
}

File: app/src/main/java/com/example/pertemuan8/ui/view/viewmodel/RencanaStudyViewModel.kt
package com.example.pertemuan8.ui.view.viewmodel

import androidx.lifecycle.ViewModel
import com.example.pertemuan8.model.RencanaStudi
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.flow.update

class RencanaStudyViewModel: ViewModel() {
    private val _krsState = MutableStateFlow(RencanaStudi())
    val krsStateUi: StateFlow<RencanaStudi> = _krsState.asStateFlow()

    fun setMataKuliah(mkPilihan: String){
        _krsState.update {stateMK -> stateMK.copy(namaMk = mkPilihan)}
    }
    fun setKelas(kelasPilihan: String) {
        _krsState.update { stateKelas -> stateKelas.copy(kelas = kelasPilihan) }
    }
    fun saveDataKRS(ls:MutableList<String>){
        _krsState.update { status -> status.copy(
            namaMk = ls[0],
            kelas = ls[1]
        )
        }
    }
}